src/NuGet.Clients/NuGetConsole.Host.PowerShell/Utils/InstalledPackageEnumerator.cs

```diff
@@ -0,0 +1,335 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft;
+using Microsoft.VisualStudio.Threading;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.PackageManagement;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.ProjectManagement;
+using NuGet.ProjectManagement.Projects;
+using NuGet.ProjectModel;
+using NuGet.Protocol.Core.Types;
+using NuGet.Resolver;
+
+namespace NuGetConsole.Host.PowerShell
+{
+    /// <summary>
+    /// Helper class collecting installed packages in all supported projects in topological order.
+    /// </summary>
+    internal class InstalledPackageEnumerator
+    {
+        private readonly ISolutionManager _solutionManager;
+        private readonly ISettings _settings;
+        private readonly Func<BuildIntegratedNuGetProject, Task<LockFile>> _getLockFileOrNullAsync;
+
+        public class PackageItem : IEquatable<PackageItem>
+        {
+            public PackageIdentity Identity { get; }
+            public string InstallPath { get; }
+
+            public PackageItem(PackageIdentity identity, string installPath)
+            {
+                Assumes.NotNull(identity);
```

Comment:
Sorry for the confusion. I reviewed usage pattern and found it always having a valid value. Then I went ahead and applied relevant changes to the constructor. So the code version is correct :)